packages/sandbox/src/setupSandboxEmbedRelay.ts

```diff
@@ -90,13 +99,43 @@ export function setupSandboxEmbedRelay({
             operationId,
             embedUrl,
           });
+        } else if (isSubscription) {
+          checkForProtocolChange = (event: MessageEvent) => {
+            if (event.data.name === EXPLORER_CHANGE_PROTOCOL) {
+              executeSubscription({
+                operation,
+                operationName,
+                variables,
+                headers,
+                embeddedIFrameElement: embeddedSandboxIFrameElement,
+                operationId,
+                embedUrl,
+                subscriptionUrl: data.subscriptionUrl,
+                protocol: event.data.protocol,
+              });
+            }
+          };
```

Comment:
We send the current protocol with every subscription request, I am confused why we need this in the first place. Can we just (in studio-ui) send a new subscription request when the protocol changes instead of a separate event? What should happen when a user has a subscription running & changes the protocol? I would imagine they need to cancel the subscription & run another one.

If we do need a separate event to change the protocol, it looks like this is only implemented in the sandbox embed relay.

Reply:
Currently, the behavior when user has a subscription running & changes the protocol, the subscription will automatically terminate and run using the new protocol.

From what I understand, the embedded explorer doesn't have a way to change the protocol after it was initially set. If it does, I can add the changes there too.


If we choose to not implement this, then it would just require users to re-run their subscription after changing protocol. This would make the code simpler, but then the behavior would be different between subscriptions in studio-ui / embed.

At one point, I had this implemented with it sending a new subscription request (SUBSCRIPTION_REQUEST) instead of CHANGE_PROTOCOL PM. The only issue with that was it resulted in extra/unnecessary SUBSCRIPTION_REQUEST PMs and extra clients being created.